### PR TITLE
[Provisioner] Accept both w/ server prefix and w/o server prefix for private docker registry

### PR DESCRIPTION
**Pull request by cblmemo**
Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [ ] Any manual or new tests for this PR (please specify below)
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

File: sky/skylet/providers/command_runner.py [hunk 1]
```diff
@@ -143,7 +143,9 @@ def run_init(self, *, as_head: bool, file_mounts: Dict[str, str],
                 docker_login_config.password,
                 docker_login_config.server,
             ))
-            specific_image = f'{docker_login_config.server}/{specific_image}'
+            server_prefix = f'{docker_login_config.server}/'
+            if not specific_image.startswith(server_prefix):
+                specific_image = f'{server_prefix}{specific_image}'
```
Comment (Michaelvll) [inline sky/skylet/providers/command_runner.py:148]: Should we handle this here or the `_with_docker_login_config` in resources? 
Pro for latter: we can assume the information provided to the node_provider or new provisioner always correct.

File: sky/task.py [hunk 2]
```diff
@@ -136,14 +136,22 @@ def _with_docker_login_config(
...
             logger.warning(f'{colorama.Fore.YELLOW}Docker login configs '
                            f'{", ".join(constants.DOCKER_LOGIN_ENV_VARS)} '
                            'are provided, but no docker image is specified '
                            'in `image_id`. The login configs will be '
                            f'ignored.{colorama.Style.RESET_ALL}')
             return resources
-        return resources.copy(_docker_login_config=docker_login_config)
+        # Already checked in extract_docker_image
+        assert len(resources.image_id) == 1
+        region = list(resources.image_id.keys())[0]
```
Comment (Michaelvll) [inline sky/task.py:149]: We should not only specifying it for a single region here, but all the regions in the `image_id`?

File: sky/task.py [hunk 3]
```diff
@@ -136,14 +136,22 @@ def _with_docker_login_config(
...
+        server_prefix = f'{docker_login_config.server}/'
```
Comment (Michaelvll) [inline sky/task.py:150]:
```suggestion
        # We automatically add the server prefix to the image name if
        # the user did not add it.
        server_prefix = f'{docker_login_config.server}/'
```

File: sky/task.py [hunk 4]
```diff
@@ -136,14 +136,22 @@ def _with_docker_login_config(
...
+        if docker_image is None:
             logger.warning(f'{colorama.Fore.YELLOW}Docker login configs '
                            f'{", ".join(constants.DOCKER_LOGIN_ENV_VARS)} '
                            'are provided, but no docker image is specified '
                            'in `image_id`. The login configs will be '
                            f'ignored.{colorama.Style.RESET_ALL}')
             return resources
-        return resources.copy(_docker_login_config=docker_login_config)
+        # Already checked in extract_docker_image
+        assert len(resources.image_id) == 1
```
Comment (Michaelvll) [inline sky/task.py:148]:
```suggestion
        assert len(resources.image_id) == 1, resources.image_id
```

File: sky/task.py [hunk 5]
```diff
@@ -136,14 +136,22 @@ def _with_docker_login_config(
...
+        docker_image = resources.extract_docker_image()
+        if docker_image is None:
             logger.warning(f'{colorama.Fore.YELLOW}Docker login configs '
                            f'{", ".join(constants.DOCKER_LOGIN_ENV_VARS)} '
                            'are provided, but no docker image is specified '
                            'in `image_id`. The login configs will be '
                            f'ignored.{colorama.Style.RESET_ALL}')
             return resources
-        return resources.copy(_docker_login_config=docker_login_config)
+        # Already checked in extract_docker_image
```
Comment (Michaelvll) [inline sky/task.py:147]: Could you remind me why we do not allow multiple regions in `image_id` when docker is specified?

**Review by Michaelvll — COMMENTED** [review]
Thanks for the update @cblmemo! LGTM with a question.

File: sky/task.py (hunk 5 above)
Comment (cblmemo) [reply sky/task.py:147]: The mix of image and docker image seems possible too - just using the corresponding image when we `make_deploy_resources_variables` will work iiuc. Let me file an issue to track on this

**Review by Michaelvll — APPROVED** [review]

**Pull request merged**